pyglet/util.py

```diff
@@ -16,25 +18,40 @@ def asbytes(s):
         return bytes(s)
 
 
-def asbytes_filename(s):
-    if isinstance(s, bytes):
+def asbytes_filename(s: Union[str, Buffer]) -> Optional[bytes]:
+    if isinstance(s, (bytes, bytearray, memoryview)):
         return s
     elif isinstance(s, str):
         return s.encode(encoding=sys.getfilesystemencoding())
 
+    return None
 
-def asstr(s):
+def asstr(s: Optional[Union[str, Buffer]]) -> str:
     if s is None:
         return ''
     if isinstance(s, str):
         return s
-    return s.decode("utf-8")
+    return s.decode("utf-8")  # type: ignore
+
+
+# Keep these outside of the function since we don't need to re-define
+# the function each time we make a call since no state is persisted.
+def _debug_print_real(arg: str) -> bool:
+    print(arg)
+    return True
+
+
+def _debug_print_dummy(arg: str) -> bool:
+    return True
+
 
+def debug_print(pyglet_option_name: str = 'debug') -> DebugPrintCallable:
+    """Get a debug print callable based on a pyglet option name.
 
-def debug_print(enabled_or_option='debug'):
-    """Get a debug printer that is enabled based on a boolean input or a pyglet option.
-    The debug print function returned should be used in an assert. This way it can be
-    optimized out when running python with the -O flag.
```

Comment:
The original description reads more clearly to me. Perhaps the reason why we are using an `assert` in the first place might need more explanation, as we are essentially hijacking the intended behavior here.
The fact that it always returns True is just a way to ensure that it never gives an assertion error. 

We use an assert here because:
1. It allows us to put print statements on one line, instead of adding `if _debug_enabled: ...` statements everywhere. 
2. When Python is run with -O (as when frozen), asserts are automatically optimized out. 

I can update this myself if you like.

Reply:
I've tried to include both the original summary line and your phrasing here. Feel free to make tweaks. :+1: